### PR TITLE
Add a usecase to create multiple domains/clusters and lifecycle tests

### PR DESCRIPTION
**Pull request by sankarpn**
Tests related to large capacity domain and multiple clusters.
- testCreateDomains
   * Test brings up new domains and verifies it can successfully start by doing the following.
   * 
   * a. Creates new WebLogic domains using offline WLST in persistent volume. 
   * b. Creates domain resource and deploys in Kubernetes cluster. 
   * c. Verifies the servers in the new WebLogic domain comes up.
-testCreateNewClusters
* Test creates new clusters and verifies it can successfully start by doing the following. 
   * 
   * a. Creates new WebLogic static clusters using online WLST. 
   * b. Patch the Domain Resource with clusters 
   * c. Update the introspectVersion version 
   * d. Verifies the servers in the new WebLogic cluster comes up.
   * e. Repeat the above cycle for a number of clusters.
   * Bug - OWLS-102898
-testCreateNewClustersDontStart
* Test creates new clusters in shutdown state and verifies it can successfully start after patching the domain with
   * new introspectVersion string 
   * 
   * a. Creates new WebLogic static clusters using online WLST. 
   * b. Patch the Domain Resource with clusters.
   * c. Update the introspectVersion version.
   * d. Verifies the servers in the new WebLogic clusters comes up without affecting any of the running servers on 
   * pre-existing WebLogic cluster.
-testRestartClusters
 * Test shuts down all existing clusters and starts up. 
   * 
   * a. Shutdowns all cluster using serverStartPolicy NEVER. 
   * b. Patch the Domain Resource with cluster serverStartPolicy IF_NEEDED. 
   * c. Verifies the servers in the domain cluster comes up.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13214/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13213/

### CODE REVIEW

**Review by xiancao — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+  private static final int t3ChannelPort = getNextFreePort();
+  private static final String wlSecretName = "weblogic-credentials";
+  private static String wlsUserName = ADMIN_USERNAME_DEFAULT;
+  private static String wlsPassword = ADMIN_PASSWORD_DEFAULT;
+
+  private static String adminSvcExtHost = null;
+  private static String clusterRouteHost = null;
+
+  // create standard, reusable retry/backoff policy
+  private static final ConditionFactory withStandardRetryPolicy
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java:123]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+    logger.info("Assign a unique namespaces for WebLogic domains");
+    domainNamespaces = namespaces.subList(1, numOfDomains + 1);
+    domainNamespace = namespaces.get(numOfDomains + 1);
+
+    // install operator and verify its running in ready state
+    installAndVerifyOperator(opNamespace, namespaces.subList(1, 50).toArray(new String[0]));
+
+    // build the clusterview application
+    Path targetDir = Paths.get(WORK_DIR,
+        ItIntrospectVersion.class.getName() + "/clusterviewapp");
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java:151]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java [hunk 3]
```diff
@@ -0,0 +1,707 @@
...
+        checkPodReadyAndServiceExists(clusterManagedServerPodNamePrefix + j, domainUid, domainNamespaces.get(i));
+      }
+
+      List<String> managedServerNames = new ArrayList<>();
+      for (int j = 1; j <= clusterReplicaCount; j++) {
+        managedServerNames.add(clusterManagedServerNameBase + j);
+      }
+      //verify admin server accessibility and the health of cluster members
+      deployAndVerifyMemberHealth(adminServerPodName, managedServerNames, domainNamespaces.get(i));
+      shutdownDomain(domainUid, domainNamespaces.get(i));
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java:213]: fixed. commented the shutdown call

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+        checkPodReadyAndServiceExists(clusterManagedServerPodNamePrefix + j, domainUid, domainNamespaces.get(i));
+      }
+
+      List<String> managedServerNames = new ArrayList<>();
+      for (int j = 1; j <= clusterReplicaCount; j++) {
+        managedServerNames.add(clusterManagedServerNameBase + j);
+      }
+      //verify admin server accessibility and the health of cluster members
+      deployAndVerifyMemberHealth(adminServerPodName, managedServerNames, domainNamespaces.get(i));
+      //shutdownDomain(domainUid, domainNamespaces.get(i));
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java:202]: delete the commented out line

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java [hunk 5]
```diff
@@ -0,0 +1,696 @@
...
+    deployAndVerifyMemberHealth(adminServerPodName, managedServerNames, domainNamespace);
+
+  }
+
+  /**
+   * Test brings up new domains and verifies it can successfully start by doing the following.
+   * 
+   * a. Creates new WebLogic domains using offline WLST in persistent volume. 
+   * b. Creates domain resource and deploys in Kubernetes cluster. 
+   * c. Verifies the servers in the new WebLogic domain comes up.
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLargeCapacityDomainsClusters.java:170]: Verifies the servers in the new WebLogic domain comes up. -> Verifies the servers in the new WebLogic domain come up.

**Review by xiancao — APPROVED** [review]
minor comments

**Review by anpanigr — APPROVED** [review]

**Pull request merged**